src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -138,7 +141,7 @@ public static async Task<string> GetDigestTokenForCredential(NetworkCredential c
 
             // Calculate response
             string a1 = credential.UserName + ":" + realm + ":" + credential.Password;
-            if (algorithm.IndexOf("sess") != -1)
+            if (algorithm.EndsWith("sess", StringComparison.OrdinalIgnoreCase))
```

Comment:
The logic of these two checks is different beyond just case-sensitivity - is this deliberate?

Reply:
Yes, it is deliberate. In practice it doesn't matter because of the previous check that restricts it to 4 known values (`md5`, `md5-sess`, `sha256` and `sha256-sess`).